### ISSUE TITLE
Updating a card source expiration date

### ISSUE THREAD

**Issue by johnnydee8** [stripe/stripe-php#445]
I am trying to update a card source expiration date, following the docs to the letter https://stripe.com/docs/sources/cards#updating-a-card-source-expiration-date

This appears to be successful (not receiving any error messages back), however after checking the source no changes are being made. 

I am not sure what causes this - my code seems fine (copy-paste from the docs above). The same source cannot be edited in the dashboard either - would that mean that Stripe does not allow to modify expiration date any more? Or would that be a problem with the package itself?

Currently used version v4.13.0 / api version 2018-02-06

Does anyone else experience this issue? Any idea what's causing this / are there any workarounds?

**Comment by remi-stripe** [comment]
@johnnydee8 I have tried the following code with your exact version of the library and it works without issue for me:

```
$source = \Stripe\Source::create(array(
  "type" => "card",
  "card" => array(
    "number" => "4242424242424242",
    "exp_month" => "09",
    "exp_year" => "2025",
  ),
  "owner" => array(
    "email" => "my_email@stripe.com",
    "name" => "my_name",
    "address" => array(
        "line1" => "my_line1",
        "line2" => "my_line2",
        "city" => "my_city",
        "postal_code" => "my_postal_code",
        "state" => "my_state",
        "country" => "my_country",
    )
  )
));

$source->card["exp_month"] = "01";
$source->card["exp_year"] = "2020";
$source->save();

$sourceRetrieved = \Stripe\Source::retrieve($source->id);
if($source->card->exp_year != 2020) {
  echo "The source has not been updated";
} else {
  echo "Source updated successfully";
}
```

I would recommend reaching out to our support team and providing the exact code and example source that you are trying to update. You can contact them here: https://support.stripe.com/email

**Comment by johnnydee8** [comment]
@remi-stripe check statement is incorrect - can you please fix and check again? 

last if statement should use $sourceRetrieved not $source - for me it does not update the source.

**Comment by remi-stripe** [comment]
🤦‍♂️ 

So you're right the code was incorrect. Once I used the right code, it failed obviously. And then I tested with the latest version of our library (6.1.0) and it succeeded.

I then looked at our changelog and confirmed that this is a bug we fixed on 5.7.0 as you can see [here]( https://github.com/stripe/stripe-php/blob/master/CHANGELOG.md#570---2017-11-28)

You'll need to upgrade the version of our library you use to properly update a Source's expiration date
